esphome/components/seeed_mr60fda2/seeed_mr60fda2.cpp

```diff
@@ -0,0 +1,437 @@
+      float height_threshold_float;
+      memcpy(&height_threshold_float, &current_height_threshold_int_, sizeof(float));
+      select_index_ = find_nearest_index_(height_threshold_float, HEIGHT_THRESHOLD, 7);
+      this->height_threshold_select_->publish_state(HEIGHT_THRESHOLD_STR[select_index_]);
+      this->current_sensitivity_ =
+          (static_cast<uint32_t>(current_data_buf_[11]) << 24) | (static_cast<uint32_t>(current_data_buf_[10]) << 16) |
+          (static_cast<uint32_t>(current_data_buf_[9]) << 8) | static_cast<uint32_t>(current_data_buf_[8]);
+      select_index_ = find_nearest_index_(this->current_sensitivity_, SENSITIVITY, 3);
+      this->sensitivity_select_->publish_state(SENSITIVITY_STR[select_index_]);
+      ESP_LOGD(TAG, "Mounting height: %.2f, Height threshold: %.2f, Sensitivity: %u", install_height_float,
+               height_threshold_float, this->current_sensitivity_);
+      this->current_frame_locate_ = LOCATE_FRAME_HEADER;
+      break;
+    default:
+      break;
+  }
+}
+
+// Sending data frames
+void MR60FDA2Component::send_query_(uint8_t *query, size_t string_length) { this->write_array(query, string_length); }
+
+/**
+ * @brief Convert a float value to a byte array.
+ *
+ * This function converts a float value to a byte array.
+ *
+ * @param value The float value to convert.
+ * @param bytes The byte array to store the converted value.
+ */
+void MR60FDA2Component::float_to_bytes_(float value, unsigned char *bytes) {
+  union {
+    float float_value;
+    unsigned char byte_array[4];
+  } u;
+
+  u.float_value = value;
+  memcpy(bytes, u.byte_array, 4);
+}
+
+/**
+ * @brief Convert a 32-bit unsigned integer to a byte array.
+ *
+ * This function converts a 32-bit unsigned integer to a byte array.
+ *
+ * @param value The 32-bit unsigned integer to convert.
+ * @param bytes The byte array to store the converted value.
+ */
+void MR60FDA2Component::int_to_bytes_(uint32_t value, unsigned char *bytes) {
+  bytes[0] = value & 0xFF;
+  bytes[1] = (value >> 8) & 0xFF;
+  bytes[2] = (value >> 16) & 0xFF;
+  bytes[3] = (value >> 24) & 0xFF;
+}
+
+// Send Heartbeat Packet Command
+void MR60FDA2Component::set_install_height(uint8_t index) {
+  uint8_t send_data[13] = {0x01, 0x00, 0x00, 0x00, 0x04, 0x0E, 0x04, 0xF0, 0x00, 0x00, 0x00, 0x00, 0x00};
+  uint8_t data_frame[4] = {0x00, 0x00, 0x00, 0x00};
+
+  float_to_bytes_(INSTALL_HEIGHT[index], &send_data[8]);
+
+  for (int i = 0; i < 4; i++) {
+    data_frame[i] = send_data[i + 8];
+  }
+
+  send_data[12] = calculate_checksum_(data_frame, 4);
+  this->send_query_(send_data, 13);
+  ESP_LOGD(TAG,
+           "SEND INSTALL HEIGHT FRAME: 0x%02x 0x%02x 0x%02x 0x%02x 0x%02x 0x%02x 0x%02x 0x%02x 0x%02x 0x%02x 0x%02x "
+           "0x%02x 0x%02x",
+           send_data[0], send_data[1], send_data[2], send_data[3], send_data[4], send_data[5], send_data[6],
+           send_data[7], send_data[8], send_data[9], send_data[10], send_data[11], send_data[12]);
```

Comment:
Please use V (verbose) logging for things like this. We also have a helper (`format_hex_pretty`) to convert arrays of data to hex strings
```suggestion
  ESP_LOGV(TAG, "SEND INSTALL HEIGHT FRAME: %s", format_hex_pretty(send_data, 13).c_str();
```